### DISCUSSION TITLE
I need help with Promises / Promise.all

### DISCUSSION THREAD

**Discussion by JuanDavila1101** [nss-evening-cohort-14/evening-client-side#306] [Capstones!] [answered]
I need help understanding and implementing for what I think I need Promises/ Promise.all

I one of my data examples is 
I am trying to use the returned FirebaseKey = response.data.name from:
```
const addSite = (siteObject) => new Promise((resolve, reject) => {
  axios.post(`${dbURL}/sites.json`, siteObject)
    .then((response) => {
      const body = { siteID: response.data.name };
      axios.patch(`${dbURL}/sites/${response.data.name}.json`, body)
        .then(() => console.warn('this is the firebase key to the new site:', response.data.name),
          resolve(response.data.name))
        .catch((error) => reject(error));
    });
});
```

if possible into something like this:
```
const addWorkerToSite = (workerSiteObject, siteID) => new Promise((resolve, reject) => {
  const workerSiteObjectTemp = ({ ...workerSiteObject, siteID });
  debugger;
  axios.post(`${dbURL}/workerInSite.json`, workerSiteObjectTemp)
    .then(() => resolve())
    .catch((error) => reject(error));
});
```

I see relatable tickets:
https://github.com/nss-evening-cohort-14/evening-client-side/discussions/216
https://github.com/nss-evening-cohort-14/evening-client-side/discussions/185
https://github.com/nss-evening-cohort-14/evening-client-side/discussions/215

I have searched in google:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all
https://www.freecodecamp.org/news/promise-all-in-javascript-with-example-6c8c5aea3e32/
https://stackoverflow.com/questions/42210200/adding-a-promise-to-promise-all

my code references are:
https://github.com/JuanDavila1101/workersonsite/blob/main/src/helpers/data/sitesData.js
https://github.com/JuanDavila1101/workersonsite/blob/main/src/helpers/data/WorkerInSiteData.js

**Comment by drteresavasquez** [comment]
@JuanDavila1101 what are you trying to accomplish?

**Reply by JuanDavila1101** [reply]
I am trying to try to resolve the adding first mainly, I am want to add all selected workers to a new site and use the returned new firebaseKey to append the new relationship for the join table. I was trying to use the resolve() and return the firebaseKey and then save it to a useState that I could then cycle through and array of selected workers and then use, the new key and workersIDs to make new individual adds. but I am having trouble returning the new firebase key as well.

**Comment by TrinityChristiana** [comment]
We talked though the logic and they are going to report back here after they’ve tried implementing what we talked about.

**Reply by JuanDavila1101** [reply]
THANK YOU SO MUCH!!!
I was able to the the promise.all like this

**Comment by JuanDavila1101** [comment, marked as answer]
THANK YOU SO MUCH!!!

Implementing Promise.all.
I was able to pass the array of workers form the site and able to append the new firebaseKey of the new site onto the call for the new workers and sites relationship in order to create a new relationship of the worker and the site.

`const addSite = (siteObject, uid, workersInSiteChanges) => new Promise((resolve, reject) => {
  axios.post(`${dbURL}/sites.json`, siteObject)
    .then((response) => {
      const body = { siteID: response.data.name };

      const addingWorkersToSite = workersInSiteChanges.map((workerinfo) => addWorkerToSite({ ...workerinfo, siteID: response.data.name }));

      const patchingTheNewSite = axios.patch(`${dbURL}/sites/${response.data.name}.json`, body);

      Promise.all([addingWorkersToSite, patchingTheNewSite])
        .then(() => {
          getSites(uid).then((sitesArray) => resolve(sitesArray));
        });
    }).catch((error) => reject(error));
});`



`const addWorkerToSite = (workerSiteObject) => new Promise((resolve, reject) => {
  console.warn(`${dbURL}/workerInSite.json`);
  axios.post(`${dbURL}/workerInSite.json`, workerSiteObject)
    .then(() => resolve())
    .catch((error) => reject(error));
});`


Thank you!!!!!
